### DISCUSSION TITLE
openshift-install create error getting S3 bucket (...) policy: AccessDenied and documentation policies problems 4.11.0-0.okd-2022-10-28-153352

### DISCUSSION THREAD

**Discussion by apmlima** [okd-project/okd#1385] [General]
$ openshift-install create cluster --dir t9
level=info msg=Credentials loaded from default AWS environment variables
level=info msg=Consuming Install Config from target directory
level=info msg=Skipping quota checks
level=info msg=Creating infrastructure resources...
level=error
level=error msg=Error: error getting S3 bucket (t9-zfrhn-bootstrap) policy: AccessDenied: Access Denied
level=error msg=	status code: 403, request id: 7XFHZW5RHKA83EA7, host id: 7wt0sb/lxFBvzezNfGDIA3v25os/FRRfMAM1Nk+EKI7BElT/oXwJWNBwj2eTSu9F4O6uS4cbQjc=
level=error
level=error msg=  with aws_s3_bucket.ignition,
level=error msg=  on main.tf line 47, in resource "aws_s3_bucket" "ignition":
level=error msg=  47: resource "aws_s3_bucket" "ignition" {
level=error
level=error msg=failed to fetch Cluster: failed to generate asset "Cluster": failure applying terraform for "bootstrap" stage: failed to create cluster: failed to apply Terraform: exit status 1
level=error
level=error msg=Error: error getting S3 bucket (t9-zfrhn-bootstrap) policy: AccessDenied: Access Denied
level=error msg=	status code: 403, request id: 7XFHZW5RHKA83EA7, host id: 7wt0sb/lxFBvzezNfGDIA3v25os/FRRfMAM1Nk+EKI7BElT/oXwJWNBwj2eTSu9F4O6uS4cbQjc=
level=error
level=error msg=  with aws_s3_bucket.ignition,
level=error msg=  on main.tf line 47, in resource "aws_s3_bucket" "ignition":
level=error msg=  47: resource "aws_s3_bucket" "ignition" {
level=error

Maybe version 4.11 needs a AWS policy I am missing? Think previous versions done a policies check.
Also found errors on the documentation regarding install user needed policies that making very hard to fix this:
Invalid Action: The action s3:HeadBucket does not exist.
Invalid Action: The action s3:GetBucketReplication does not exist.
Invalid Action: The action s3.GetBucketPolicy does not exist.

Can someone please help? Vadim @vrutkovs ?
Thanks!

Pedro

4.11.0-0.okd-2022-10-28-153352

100%

https://drive.google.com/file/d/1vHl8wFgrvb3dVZZ6Kpmmuhvehx7nUraX/view?usp=sharing

**Comment by vrutkovs** [comment]
install log is empty. See https://docs.openshift.com/container-platform/4.11/installing/installing-troubleshooting.html#installing-getting-debug-information_installing-troubleshooting also for more verbosity

**Comment by LorbusChris** [comment]
The AWS account is missing permissions. It's not really an OKD issue, and not actionable for us.
This is the same issue as https://github.com/okd-project/okd-scos/issues/3 for OKD/SCOS

**Comment by apmlima** [comment]
@vrutkovs , Thanks for the fast reply.
I just opened the file and there is a 326k log.
BTW, I am creating 4.6-4.10 clusters with no permission policies issues.
Well, here it is a new log of a new cluster I created in debug mode with the same error.
https://drive.google.com/file/d/1C5JhbhbTN0aD3RkTflWEmOIauh_c8B6W/view?usp=sharing

@LorbusChris , several permissions/AWS policies specified on the documentations don't even exist and is because of that I opened the issue because I don't know what you need for your openshift-install create cluster and or you have a problem with your binaries or with the documentation.

**Comment by aleskandro** [comment]
Hello @apmlima, I just tried right now and it works as expected:

1. Downloaded the latest installer from https://github.com/okd-project/okd-scos/releases/download/4.12.0-0.okd-scos-2022-10-25-053756/openshift-client-linux-4.12.0-0.okd-scos-2022-10-25-053756.tar.gz (based on your initial issue)
2. Created a IAM user with the `AdministratorAccess` policy attached to it
3. Set the IAM user in `~/.aws/credentials` like
```
[test-okd]
aws_access_key_id = ...
aws_secret_access_key =...
```
5. `export AWS_PROFILE=test-okd`
6.
 ```
./openshift-install version
./openshift-install 4.12.0-0.okd-scos-2022-10-25-053756
built from commit 14d496fdaec571fa97604a487f5df6a0433c0c68
release image quay.io/okd/scos-release@sha256:982290ee4e0672dcf435148147d35d252c1be4263c3567d24c098575ef0d2e58
release architecture amd64
aleskandro@dujour Downloads % ./openshift-install create install-config --dir=test
```
7. 
```
./openshift-install create install-config --dir=test
? SSH Public Key /home/aleskandro/.ssh/id_rsa.pub
? Platform aws
INFO Credentials loaded from the "test-okd" profile in file "/home/aleskandro/.aws/credentials" 
? Region us-east-2
? Base Domain ****my.base.domain.tld***
? Cluster Name my-test-221101
? Pull Secret [? for help] ********************************************************************************************************************************************************************INFO Install-Config created in: test     
```
8. `./openshift-install create cluster --dir=test --log-level=debug`
The documentation clearly states that the IAM User used to install the cluster on AWS  has to have the AdministratorAccess policy bound to it.

**Is the IAM User you are using bound to the `AdministratorAccess` policy?** If not, please retry following the section `Creating the IAM user` in the [documentation](https://docs.okd.io/4.11/installing/installing_aws/installing-aws-account.html#installation-aws-iam-user_installing-aws-account) again:

> 1. Specify the IAM user name and select Programmatic access.
> 2. Attach the AdministratorAccess policy to ensure that the account has sufficient permission to create the cluster. This policy provides the cluster with the ability to grant credentials to each OpenShift Container Platform component. The cluster grants the components only the credentials that they require.
******While it is possible to create a policy that grants the all of the required AWS permissions and attach it to the user, this is not the preferred option. *******The cluster will not have the ability to grant additional credentials to individual components, so the same credentials are used by all components.
> 4. Optional: Add metadata to the user by attaching tags.
> 5. Confirm that the user name that you specified is granted the AdministratorAccess policy.
> 6. Record the access key ID and secret access key values. You must use these values when you configure your local machine to run the installation program.


To avoid other confusion, be also sure that the cluster name you are using is not the same as in one of the previous tests (and remember to destroy clusters, even if they didn't complete the installation, they could left some dirty resources in your AWS account)

**Reply by apmlima** [reply]
@aleskandro , I also can create clusters with my Admin account with no problems.
Enterprises for security reasons need to limit users access rights to only what are strictly required to do their jobs, we use an AWS account for cluster create purposes with the needed rights (policies) we get from your documentation. Your documentation even mentions "it is possible to create a policy that grants the all of the required AWS permissions and attach it to the user".
I am doing this for several years from 4.6-4.10 with no problems and you changed openshift-install in 4.11 needing more permissions without updating the documentation that have a few inexistent ones I already told @LorbusChris . Due to that I am telling you about this issue and request your help.
